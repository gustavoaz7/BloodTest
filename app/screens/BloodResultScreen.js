import React, { Component } from 'react';
import { View, FlatList, Alert } from 'react-native';
import axios from 'axios';
import styled from 'styled-components/native';
import Loading from '../components/Loading';
import BloodResultItem from '../components/BloodResultItem';

export default class BloodResultScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      results: [],
      error: false,
      loading: true,
    };
  }

  componentDidMount() {
    this.loadResults();
  }

  loadResults = async () => {
    try {
      const { data: results } = await axios.get('http://192.168.0.13:4000/results');
      this.setState({ results, loading: false, error: false });
    } catch (e) {
      this.setState({ results: [], error: true, loading: false });
      console.warn('BloodListScreen load tests error - ', e);
    }
  };

  handleError() {
    Alert.alert(
      'Something went wrong :(',
      'There was a problem connecting to our servers.\n\nPlease try again in a few seconds.',
      [
        {
          text: 'Try again',
          onPress: this.loadResults,
        },
      ],
    );
  }

  keyExtractor(item) {
    return item.id;
  }

  renderItem({ item }) {
    return <BloodResultItem item={item} />;
  }

  renderSeparator() {
    return <Separator />;
  }

  render() {
    const { results, loading, error } = this.state;
    if (loading) return <Loading />;
    if (error) {
      this.handleError();
      return null;
    }
    return (
      <FlatListUI
        data={results}
        renderItem={this.renderItem}
        keyExtractor={this.keyExtractor}
        ItemSeparatorComponent={this.renderSeparator}
      />
    );
  }
}

const FlatListUI = styled(FlatList)`
  padding: 12px;
  background-color: 'rgba(0, 0, 0, 0.05)';
`;

const Separator = styled(View)`
  height: 12px;
`;
