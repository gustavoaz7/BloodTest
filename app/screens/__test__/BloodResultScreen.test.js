import React from 'react';
import { Button } from 'react-native';
import { shallow } from 'enzyme';
import BloodResultScreen from '../BloodResultScreen';

describe('BloodResultScreen', () => {
  it('renders without crashing', () => {
    shallow(<BloodResultScreen />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BloodResultScreen />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handle navigate press', () => {
    const goBackSpy = jest.fn();
    const navigation = { goBack: goBackSpy };
    const wrapper = shallow(<BloodResultScreen navigation={navigation} />);

    wrapper
      .find(Button)
      .at(0)
      .props()
      .onPress();

    expect(goBackSpy).toHaveBeenCalled();
  });
});
