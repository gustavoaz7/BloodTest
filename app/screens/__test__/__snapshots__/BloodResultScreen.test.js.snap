// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BloodResultScreen matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    BloodResultScreen
  </Text>
  <Button
    onPress={[Function]}
    title="Back to blood list"
  />
</View>
`;
